### PR TITLE
Refactors and fixes

### PR DESCRIPTION
**Pull request by syrusakbary**
This PR:
* [x] Updates xterm to use version 4
* [x] Adds a sync option to `print` (so the xterm will write things synchronously)
* [x] Fix Firefox prompting issue

firefox-prompt

### CODE REVIEW

File: packages/wasm-terminal/lib/wasm-terminal.ts [hunk 1]
```diff
@@ -35,11 +32,13 @@ export default class WasmTerminal {
     this.wasmTerminalConfig = new WasmTerminalConfig(config);
 
     // Create our xterm element
-    this.xterm = new Terminal();
+    this.xterm = new Terminal({
+      // rendererType: 'dom'
+    });
     // tslint:disable-next-line
-    this.pasteEvent = this.xterm.on("paste", this.onPaste);
+    // this.pasteEvent = this.xterm.on("paste", this.onPaste);
```
Comment (torch2424) [inline packages/wasm-terminal/lib/wasm-terminal.ts]: Is this handled by xterm now? 🤔

File: packages/wasm-terminal/lib/wasm-terminal.ts [hunk 2]
```diff
@@ -62,7 +63,7 @@ export default class WasmTerminal {
     this.wasmTty = new WasmTty(this.xterm);
     this.wasmShell = new WasmShell(this.wasmTerminalConfig, this.wasmTty);
     // tslint:disable-next-line
-    this.dataEvent = this.xterm.on("data", this.wasmShell.handleTermData);
+    this.dataEvent = this.xterm.onData(this.wasmShell.handleTermData);
```
Comment (torch2424) [inline packages/wasm-terminal/lib/wasm-terminal.ts]: Yay! Glad we're moving to the newer API 😄

File: packages/wasm-terminal/lib/wasm-tty/wasm-tty.ts [hunk 3]
```diff
@@ -110,9 +110,20 @@ export default class WasmTTY {
...
-    const normInput = message.replace(/[\r\n]+/g, "\n");
-    this.xterm.write(normInput.replace(/\n/g, "\r\n"));
+  print(message: string, sync?: boolean) {
+    const normInput = message.replace(/[\r\n]+/g, "\n").replace(/\n/g, "\r\n");
+    if (sync) {
+      // Force a reload in xterm
+      //@ts-ignore
+      this.xterm._core.writeSync(normInput);
+      //@ts-ignore
+      this.xterm._core._renderService._renderer._runOperation(renderer =>
```
Comment (torch2424) [inline packages/wasm-terminal/lib/wasm-tty/wasm-tty.ts]: Nice solution! 😄 🎉

File: packages/wasm-terminal/lib/wasm-terminal.ts (hunk 1 above)
Comment (syrusakbary) [reply packages/wasm-terminal/lib/wasm-terminal.ts]: Yes, I couldn't see any paste methods there. And copy/paste still worked locally :)

**Comment by syrusakbary** [comment]
It finally passed the tests 🎉
Merging

**Pull request merged**